archive/reader.go

```diff
@@ -1,175 +1,328 @@
-// Package archive is used to parse a .zip file retrieved
-// by the API.
-//
-// Here is the content of an archive retried on the tablet as example:
-// 384327f5-133e-49c8-82ff-30aa19f3cfa4.content
-// 384327f5-133e-49c8-82ff-30aa19f3cfa4//0-metadata.json
-// 384327f5-133e-49c8-82ff-30aa19f3cfa4//0.rm
-// 384327f5-133e-49c8-82ff-30aa19f3cfa4.pagedata
-// 384327f5-133e-49c8-82ff-30aa19f3cfa4.thumbnails/0.jpg
-//
-// As the .zip file from remarkable is simply a normal .zip file
-// containing specific file formats, this package is only a wrapper
-// around the standard zip package that will follow the same
-// code architecture and that will help gathering one of
-// those specific files more easily.
 package archive
 
 import (
 	"archive/zip"
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"io"
+	"io/ioutil"
 	"path/filepath"
-	"sort"
 	"strconv"
 	"strings"
+
+	"github.com/juruen/rmapi/encoding/rm"
 )
 
-// A Page represents a note page.
-type Page struct {
-	Data      *zip.File
-	Metadata  *zip.File
-	Thumbnail *zip.File
+// Read fills a File struct parsing a Remarkable archive file.
+func (z *Zip) Read(r io.Reader) error {
```

Comment:
I was wondering if `Read`/`Write` were good names here. The real purpose of this function is to parse/decode data from a reader and to load it into a predefined struct. To me, it is a bit like the `json.Decoder` in the standard library. 

`func NewDecoder(r io.Reader) *Decoder` and `func (dec *Decoder) Decode(v interface{}) error`

It takes an `io.Reader` as well and serves the purpose of unmarshaling from a stream. In some kind, here we are unmarshaling from a `zip` format (according to Wikipedia: "unmarshaling refers to the process of transforming a representation of an object that was used for storage or transmission to a representation of the object that is executable").

It is an after thought but maybe it would be more appropriate to call them `Decode`/`Encode` if wanting to stay consistent with the standard library (or maybe `Marshal`/`Unmarshal` but in the context of json, these corresponding methods act upon a `[]byte` and not a stream).

Maybe you have an opinion on this?

Reply:
I don't have a strong opinion here to be honest. If you are cool with it, we could leave it as it, and if at some point in the future we decide there's a more idiomatic or sensible way of naming this function, we can just go and change it.